components/server/src/workspace/workspace-starter.ts

```diff
@@ -332,46 +339,84 @@ export class WorkspaceStarter {
         }
     }
 
-    private async buildImageAndStartWorkspace(
-        ctx: TraceContextWithSpan,
-        user: User,
-        workspace: Workspace,
-        instance: WorkspaceInstance,
-        envVars: ResolvedEnvVars,
-    ): Promise<StartWorkspaceResult> {
-        const { span } = ctx;
+    public reconcileWorkspaceStart(_ctx: TraceContext, instanceId: string, user: User, workspace: Workspace) {
+        const ctx = TraceContext.childContext("reconcileWorkspaceStart", _ctx);
+
+        // We try to acquire a mutex here, which we intend to hold until the workspace start request is sent to ws-manager.
+        // In case this container dies for whatever reason, the mutex is eventually released, and the instance can be picked up
+        // by another server process (cmp. WorkspaceStartController).
+        this.redisMutex
```

Comment:
what do you think of using `workspace_start_controller` also to shield using teh redisMutex? I.e. we wouldn't rollout this mutex on the hot path to 100% of our users immediately.

Reply:
Yeah, was thinking in the same direction: Basically this PR contains three changes:
 1. do away with he "sometimes-sync" path in startWorkspace
 1. introduce the mandatory mutex on startWorkspace
 1. the control loop (behind `workspace_start_controller`)

Happy to guard 2. and 3. behind a single flag as you proposed. :+1:

Reply:
Done.